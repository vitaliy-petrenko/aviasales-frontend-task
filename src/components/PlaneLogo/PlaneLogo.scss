@import '../../styles/abstract/mixins';

@keyframes earth-spin {
  from {
    transform: translate3d(0, 0, 0) rotate(0)
  }

  to {
    transform: translate3d(0, 0, 0) rotate(360deg)
  }
}

.plane-logo {
  @include make-ratio;

  width: 60px;
  position: relative;
  border-radius: 50%;
  box-shadow: 0 8px 11px rgba(55, 104, 142, 0.25);
  background: var(--color-primary);
}

.plane-logo__plane-wrap {
  width: 50px;
  height: 51px;
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate3d(-50%, -50%, 0)
}

.plane-logo__earth {
  background: url('../../assets/icons/earth.svg');
  @include overlay;
}

.plane-logo__plane {
  background: url('../../assets/icons/plane.svg') no-repeat 0 0;
  @include overlay;
}
