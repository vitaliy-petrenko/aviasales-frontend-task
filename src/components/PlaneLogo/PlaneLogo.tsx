import React from 'react'
import './PlaneLogo.scss'
import { motion } from 'framer-motion'

interface IProps {
  animated?: boolean
}

const PlaneLogo: React.FC<IProps> = ({ animated = false }) => {
  return (
    <div className='plane-logo'>
      <motion.div
        className='plane-logo__earth'
        animate={
          animated ? {
            scale: [1, 1.1, 1, 1.1, 1], rotate: [0, 180],
          } : {
            scale: 1, rotate: 0
          }
        }
        transition={
          animated ? {
            duration: 2, loop: Infinity
          } : {
            duration: .5, loop: 0
          }
        }
      />

      <div className='plane-logo__plane-wrap'>
        <motion.div
          className='plane-logo__plane'
          animate={
            animated ? {
              scale: [1, .8, .8, 1], rotate: [0, -1, 0, -1, 0], x: [0, -1, 0, 1, 0], y: [0, 1, 0, -1, 0]
            } : {
              scale: 1, rotate: 0, x: 0, y: 0
            }
          }
          transition={
            animated ? {
              duration: 4, loop: Infinity
            } : {
              duration: .5, loop: 0
            }
          }
        />
      </div>
    </div>
  )
}

export default PlaneLogo

