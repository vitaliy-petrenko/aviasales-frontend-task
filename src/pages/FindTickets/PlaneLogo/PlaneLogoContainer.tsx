import { connect } from 'react-redux'
import PlaneLogo, { IProps } from '../../../components/PlaneLogo/PlaneLogo'
import { IState } from '../../../store/rootReducer'
import { getFetchingStatuses } from '../../../store/findTickets/selectors'

type TStateProps = IProps;

const mapStateToProps = (state: IState): TStateProps => ({
  animated: getFetchingStatuses(state).isFetching,
})

export default connect(mapStateToProps)(PlaneLogo)
