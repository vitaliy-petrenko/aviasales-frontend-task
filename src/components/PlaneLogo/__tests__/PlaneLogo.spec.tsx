import React from 'react'
import PlaneLogo from '../index'
import { shallow } from 'enzyme'

describe('PlaneLogo component', () => {
  const component = shallow(<PlaneLogo animated={false}/>)

  it('renders properly', () => {
    expect(component).toMatchSnapshot()
  })
})
