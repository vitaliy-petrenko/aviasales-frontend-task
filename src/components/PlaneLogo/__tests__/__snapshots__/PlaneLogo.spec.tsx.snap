// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlaneLogo component renders properly 1`] = `
<div
  className="plane-logo"
>
  <ForwardRef(MotionComponent)
    animate={
      Object {
        "rotate": 0,
        "scale": 1,
      }
    }
    className="plane-logo__earth"
    transition={
      Object {
        "duration": 0.5,
        "loop": 0,
      }
    }
  />
  <div
    className="plane-logo__plane-wrap"
  >
    <ForwardRef(MotionComponent)
      animate={
        Object {
          "rotate": 0,
          "scale": 1,
          "x": 0,
          "y": 0,
        }
      }
      className="plane-logo__plane"
      transition={
        Object {
          "duration": 0.5,
          "loop": 0,
        }
      }
    />
  </div>
</div>
`;
